File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int majorityElement = nums[0];
        int count =1;
        for(int i=1;i<nums.length;i++){
            if(nums[i]==majorityElement){
                count++;
                continue;
            }else{
                if(count == 0){
                    majorityElement = nums[i];
                    count= 1;
                }else
                    count--;
            }
        }

        return majorityElement;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        num = nums[0]
        cnt = 1
        for i in range(1, len(nums)):
            if nums[i] == num:
                cnt += 1
            elif cnt == 0:
                num = nums[i]
                cnt = 1
            else:
                cnt -= 1
        return num
